app/src/main/java/com/secuso/privacyfriendlycodescanner/qrscanner/ui/resultfragments/URLResultFragment.java

```diff
@@ -95,23 +95,14 @@ public void onProceedPressed(Context context) {
         if (!checked) {
             Toast.makeText(context, R.string.conform_url, Toast.LENGTH_LONG).show();
         } else {
-            String caption;
-            String qrurl3;
-            final String lowercase_qrurl = qrurl.toLowerCase();
-            if (!lowercase_qrurl.startsWith("http://") && !lowercase_qrurl.startsWith("https://")) {
-                qrurl3 = "http://" + qrurl;
-
-                Intent url = new Intent(Intent.ACTION_VIEW);/// !!!!
-                url.setData(Uri.parse(qrurl3));
-                caption = getResources().getStringArray(R.array.url_array)[0];
-                startActivity(Intent.createChooser(url, caption));
-            } else {
-                Intent url = new Intent(Intent.ACTION_VIEW);/// !!!!
-                url.setData(Uri.parse(qrurl).normalizeScheme());
-                caption = getResources().getStringArray(R.array.url_array)[0];
-                startActivity(Intent.createChooser(url, caption));
-
+            String urlForIntentData = qrurl;
+            if (!qrurl.toLowerCase().contains("://")) {
```

Comment:
This does not respect the RFC 3986 which specifies what a valid protocol-scheme looks like.
A regex to match that should be `^[a-zA-Z][a-zA-Z0-9+.-]*:.*$`, but this needs to be checked and tested.

Reply:
Thanks for improving on this! :smile: 
I'll take a look and test your changes myself in the next few days and will check if using https as default won't break any expected use case.
We'd appreciate if you'd rebase your changes onto `development` and target `development` for the PR.